collector/lib/NetworkStatusInspector.cpp

```diff
@@ -0,0 +1,177 @@
+#include "NetworkStatusInspector.h"
+
+#include <forward_list>
+#include <string>
+#include <unordered_map>
+
+#include <arpa/inet.h>
+
+#include "ConnTracker.h"
+
+namespace collector {
+
+const std::string NetworkStatusInspector::kBaseRoute = "/state/network";
+const std::string NetworkStatusInspector::kEndpointRoute = kBaseRoute + "/endpoint";
+const std::string NetworkStatusInspector::kConnectionRoute = kBaseRoute + "/connection";
+
+const std::string NetworkStatusInspector::kQueryParam_container = "container";
+
+NetworkStatusInspector::NetworkStatusInspector(const std::shared_ptr<ConnectionTracker> conntracker) : conntracker_(conntracker) {
+}
+
+static std::string IP2str(const Address& addr) {
+  switch (addr.family()) {
+    case Address::Family::IPV4: {
+      char str[INET_ADDRSTRLEN];
+      inet_ntop(AF_INET, addr.data(), str, INET_ADDRSTRLEN);
+      return str;
+    }
+    case Address::Family::IPV6: {
+      char str[INET6_ADDRSTRLEN];
+      inet_ntop(AF_INET6, addr.data(), str, INET6_ADDRSTRLEN);
+      return str;
+    }
+    default:
+      break;
+  }
+  return "unk";
+}
+
+static std::string Proto2str(L4Proto proto) {
+  switch (proto) {
+    case collector::L4Proto::TCP:
+      return "TCP";
+    case collector::L4Proto::UDP:
+      return "UDP";
+    case collector::L4Proto::ICMP:
+      return "ICMP";
+    default:
+      break;
+  }
+  return "UNKNOWN";
+}
+
+static std::string IPNet2str(const IPNet& net) {
+  return IP2str(net.address()) + (net.bits() == 0 ? "" : "/" + std::to_string(net.bits()));
+}
+
+static void SerializeEndpoint(const Endpoint& ep, Json::Value& node) {
+  node["address"] = IPNet2str(ep.network());
+  node["port"] = ep.port();
+}
+
+bool NetworkStatusInspector::handleGetEndpoints(struct mg_connection* conn, const QueryParams& queryParams) {
+  collector::AdvertisedEndpointMap endpointStates = conntracker_->FetchEndpointState(true, false);
+  Json::Value bodyRoot(Json::objectValue);
+
+  std::unordered_map<std::string, std::forward_list<collector::ContainerEndpointMap::value_type*>> byContainer;
+
+  std::optional<std::string> containerFilter = GetParameter(queryParams, kQueryParam_container);
+
+  for (auto& endpointState : endpointStates) {
+    if (!containerFilter || *containerFilter == endpointState.first.container())
+      byContainer[endpointState.first.container()].push_front(&endpointState);
```

Comment:
Can we put braces around this if statement? Not having them makes me nervous.
```suggestion
    if (!containerFilter || *containerFilter == endpointState.first.container()) {
      byContainer[endpointState.first.container()].push_front(&endpointState);
    }
```